### ISSUE TITLE
Most rooms & spaces are not shown / unusable

### ISSUE THREAD

**Issue by Zocker1999NET** [vector-im/element-web#23160]
### Steps to reproduce

1. Install Element Desktop 1.11.3 OR use https://app.element.io (same version) in Firefox
2. Log in

#### What did you expect?

To see all the spaces & rooms I joined like in nheko & Element Web/Desktop 1.11.2 & Element Android.

#### What happened instead?

I only see some rooms & spaces. If I try to search for rooms I'm joined (according to the clients), they get displayed as "Empty Room" and I get asked to join them.

#### What I tried so far?

- `Clear cache & reload`
- fully log out & re-log in
- for Desktop: `rm -rf ~/.config/Element`
- for Web: `Clear cookies and site data`
- a different PC (same OS)

#### Other info

- ~~On the same PC the same rooms are not shown for Desktop & Web and at each try, on another PC other rooms are not shown~~ they rooms hidden differ on the same PC as well, it just looked more a like.
- Fully reloading / clearing cache on other clients works fine

### Operating system

Debian GNU/Linux Testing/Unstable

### Application version

Element version: 1.11.3 Olm version: 3.2.12

### How did you install the app?

deb from https://packages.element.io/debian/

### Homeserver

matrix.org

### Will you send logs?

Yes

**Comment by Zocker1999NET** [comment]
Okay, interestingly downgrading to Element 1.11.2 & 1.11.1 and re logging in didn't helped, but I know that Element 1.11.2 was working fine before. I try to investigate what I've changed on both of my systems in the last days.

**Comment by freelock** [comment]
This happened to me as well, on Firefox with a self-hosted Element. Not sure the "empty rooms" thing is the same -- I see some empty rooms but most of the rooms I'm looking for come up in search.

Linux/Ubuntu 22.04 with Firefox loaded in a snap. I did have an OOM kill that crashed this tab. Clearing the cache did not fix it.

I do have develop.element.io open in the same account in Chrome, and that still shows all my spaces, as does Element Android.

**Comment by freelock** [comment]
After clearing the cache on the element dev version, the same spaces disappeared. So maybe this is a synapse issue?

**Comment by progval** [comment]
Hey, sorry that was because of a test I ran in a personal room, and it was broadcasted to other rooms. I reported the details to Matrix's security team and it is tracked as NVT#1537459 and NVT#1537460

**Comment by Zocker1999NET** [comment]
@progval I'm confused xD. Can you explain at least a little bit further why you think your doing caused this issue? Was this kind of an attack / security incident? Could a malicious formatted message cause this issue?

**Comment by progval** [comment]
I won't give more details as it would make it too easy to exploit, sorry. You'll have to wait for Matrix's security team to fix it

**Comment by aschrijver** [comment]
This also happened to me just now, after a reboot. Everything was fine before. Only some rooms shown and no way to get them back or sync with my old laptop that still has them. I am on Ubuntu 22.04 (fresh install) and Firefox 103.0.2. 

Element version: 1.11.3
Olm version: 3.2.12

**Comment by bdelwood** [comment]
I have experienced this on both the matrix.org homeserver, and my own, across 2 versions of element.

Element 1.11.3
Element Nightly (0.0.1-nightly.2022082501)

Synapse 1.65.0

**Comment by gi-yt** [comment]
I had the same issue on two homeservers, vern.cc and envs.net after clearing cache
```
vern:
(riot.vern.cc)
~vern element version: 1.11.3
Olm version: 3.2.12
```
```
Envs:
(develop.element.io)
Element version: 44e01823a35a-react-5aae974e93aa-js-c1160f40c2de
Olm version: 3.2.12
```

**Comment by aschrijver** [comment]
Follow-up to [the above comment](https://github.com/vector-im/element-web/issues/23160#issuecomment-1229264426). I just logged in on my Android phone where I had been logged out for a couple of weeks. I had to re-verify the session, using my old laptop for that.

In the Rooms tab I am missing 8 rooms, and they are rooms I am member of for a long time (so not ones I added while logged out on the phone). Didn't count DM rooms. My old laptop Firefox Web UI is the only place where I have an intact account it seems. Mobile phone Android app and new laptop Web UI both have corrupted Room/Space configs in different ways.

Version: 1.4.32 [40104322] (G-b10318)
Matrix SDK Version: 1.4.32 (ce28d7f8)
olm version: 3.2.12

This version is installed via F-Droid.

**Comment by ara4n** [comment]
the workaround for this is to reload the cache on element web; a proper fix will be forthcoming shortly.

**Comment by aschrijver** [comment]
Thank you @ara4n. But if you mean "Clear cache and reload", that does not work. I tried that before and just re-tried, and the OP also reloaded cache.

**Comment by turt2live** [comment]
At the time it wouldn't have worked but it should now. If not, please submit debug logs from Settings -> Help for us to investigate.

**Comment by aschrijver** [comment]
Thanks. I just sent you the logs.

**Comment by Peetz0r** [comment]
> the workaround for this is to reload the cache on element web; a proper fix will be forthcoming shortly.

> At the time it wouldn't have worked but it should now. If not, please submit debug logs from Settings -> Help for us to investigate.

I selfhost element-web. On what exact versions of element-web should this work?

**Comment by turt2live** [comment]
A proper fix is expected to go out this week, however the workaround should do the job on the latest release.

**Comment by progval** [comment]
@turt2live The workaround does not work on all rooms, see my last reply to NVT#1537459

**Comment by turt2live** [comment]
We're aware - the workaround doesn't have to be a 100% guaranteed chance of success, as much as we'd like it to be.

**Comment by turt2live** [comment]
This should be fixed as of v1.11.4 - please update asap, and find out more from our blog: https://element.io/blog/element-web-desktop-1-11-4-a-security-update-deferred-dms-and-more/

**Comment by aschrijver** [comment]
Works! Thank you @turt2live @robintown @progval for hard work on this fix :pray: Here's the [matrix announcement](https://matrix.org/blog/2022/08/31/security-releases-matrix-js-sdk-19-4-0-and-matrix-react-sdk-3-53-0) with CVE links.

PS. On both my Android apps, Element and FluffyChat, the number of rooms isn't what it should be (see [above](https://github.com/vector-im/element-web/issues/23160#issuecomment-1229817608)). I will see if there's issues related to that on respective issue trackers.
